### ISSUE TITLE
[Question] Modern VLM such as Qwen2.5-VL adaptation

### ISSUE THREAD

**Issue by dprokhorov17** [mlbio-epfl/joint-inference#2]
Hello,

Interesting work. However, in your experiments, you are using a very old (by today's definitions) VLM. Do you have an example of applying your framework to the latest generation of VLMs, such as the Qwen2.5-VL series?

A starting point for incorporating new VLMs would be great.

**Comment by agadetsky** [comment]
Dear @dprokhorov17,

Thank you for your kind words about our work!

We do have experiments with the modern GPT-4o on the ImageNet, MMMU and MMMU-Pro visual tasks (Table 4 in the paper). We will share the implementation for this experiment later.

If you would like to add support for your VLM of interest, then `unsupervised_icl_llm_gsm8k.py` should be a good starting point for you **for the open-ended tasks**. You would need to adapt the model loading, prompts preparation and answers post-processing accordingly.

If you aim to solve the close-ended tasks (e.g. classification) with the open-source model, then you would need to implement a custom reward in `rewards/rewards.py` and a custom task encoder in `encoders/`. 

Let me know if that helps.

Best,
Artyom

**Comment by dprokhorov17** [comment]
Hi @agadetsky,

my Task is a closed one 🤔. I might be completely off here, but I was aiming to conduct experiments on visual grounding for GUIs — including mobile, desktop, or web interfaces. Essentially, the idea is for a VLM to output a coordinate (or a bounding box), and the reward would be based on the Euclidean distance for points, or Distance-IoU for bounding boxes.

**Comment by agadetsky** [comment]
Dear @dprokhorov17,

First, your task seems to be an open-ended according to your description. To clarify, a close-ended task is a task for which you can enumerate the space of possible answers. For example, in classification or multiple choice questions, you have a finite set of classes or options respectively. In such case, it is better to employ unsupervised fine-tuning rather than unsupervised in-context learning to adapt your VLM to this task as we experimentally show in the paper. Thus, I recommended you to implement a custom reward in `rewards/rewards.py` and a custom task encoder in `encoders/` using your VLM of interest. These custom rewards then can be used to run unsupervised fine-tuning with `unsupervised_ft_vlm.py` on the close-ended task of interest. 

Secondly, it also seems to me that you may not need an unsupervised adaptation technique since you are telling that you already have _"a reward based on the Euclidean distance for the points"_. In such case, it is probably better to directly optimize this reward, which is out-of-the-scope of our work. In our paper, we show that in absence of ground truth rewards or ground truth answers, we can employ a foundation model itself to define a reward that can be optimized to improve a model's predictions upon the zero-shot predictions. In other words, we consider unsupervised setting, while yours seems to be supervised.

Let me know if these clarifications help you.

Best,
Artyom

**Comment by dprokhorov17** [comment]
@agadetsky Thanks for the clarification. I’ve already trained various VLMs on my task, and they worked well. I also experimented with GRPO for this type of task.

P.S. I’m just having fun exploring “alternative” training scenarios ;)
